src/Data/ProgrammesDb/EntityRepository/BroadcastRepository.php

```diff
@@ -98,4 +99,54 @@ private function setEntityTypeFilter(QueryBuilder $qb, string $type, string $bro
 
         return $qb;
     }
+
+    protected function resolveParents(array $categories): array
+    {
+        return $this->abstractResolveAncestry($categories, [$this, 'findByIds']);
+    }
+
+    private function programmeAncestryGetter(array $ids): array
+    {
+        /** @var CoreEntityRepository $repo */
+        $repo = $this->getEntityManager()->getRepository('ProgrammesPagesService:CoreEntity');
+        return $repo->findByIds($ids);
+    }
+
+    public function findAllByServiceAndDateRange(
+        $serviceId,
+        DateTimeImmutable $startDate,
+        DateTimeImmutable $endDate,
+        $limit
+    ) {
+        $startDate = $startDate->setTimezone(new DateTimeZone('UTC'));
```

Comment:
We shouldn't be manipulating dates this deep within the library - it makes it impossible for people to pass in a funky date if they actually need to. Any Date sanitization should happen as part of the Controller.